prow/cmd/branchprotector/protect.go

```diff
@@ -350,14 +357,30 @@ func (p *protector) UpdateRepo(orgName string, repoName string, repo config.Repo
 	for bn, githubBranch := range branches {
 		if branch, err := repo.GetBranch(bn); err != nil {
 			errs = append(errs, fmt.Errorf("get %s: %w", bn, err))
-		} else if err = p.UpdateBranch(orgName, repoName, bn, *branch, githubBranch.Protected, collaborators, teams); err != nil {
+		} else if err = p.UpdateBranch(orgName, repoName, bn, *branch, githubBranch.Protected, apps, collaborators, teams); err != nil {
 			errs = append(errs, fmt.Errorf("update %s from protected=%t: %w", bn, githubBranch.Protected, err))
 		}
 	}
 
 	return utilerrors.NewAggregate(errs)
 }
 
+// authorizedApps returns the list of slugs for apps that are authorized
+// to write to repositories of the org.
+func (p *protector) authorizedApps(org string) ([]string, error) {
+	appInstallations, err := p.client.ListAppInstallationsForOrg(org)
+	if err != nil {
+		return nil, err
+	}
+	var authorized []string
+	for _, a := range appInstallations {
+		if a.Permissions.Contents == string(github.Write) {
```

Comment:
So this can only manage apps that have write perms?

Reply:
Yes, if the `verify-restrictions` command line option is set.
I built it similar to the verification of collaborators which checks if they have [push or admin permissions](https://github.com/kubernetes/test-infra/blob/9d2028fd1a49ea4cf16692b87906b4bbc926c87f/prow/cmd/branchprotector/protect.go#L370-L372)